suite-native/device-manager/src/components/DeviceList.tsx

```diff
@@ -0,0 +1,176 @@
+import { Button, VStack, Box, TextDivider } from '@suite-native/atoms';
+import { TrezorDevice } from '@suite-common/suite-types';
+import { Translation } from '@suite-native/intl';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+import { DeviceItem } from './DeviceItem/DeviceItem';
+import { useDeviceManager } from '../hooks/useDeviceManager';
+import { MANAGER_MODAL_BOTTOM_RADIUS } from './DeviceManagerModal';
+
+type NavigationProp = StackToStackCompositeNavigationProps<
+    RootStackParamList,
+    RootStackRoutes.AppTabs,
+    RootStackParamList
+>;
+
+type DeviceListProps = {
+    isVisible: boolean;
+    onSelectDevice: (device: TrezorDevice) => void;
+};
+
+const ITEM_HEIGHT = 62;
+const ITEMS_SPACING = 8;
+const BUTTON_HEIGHT = 48;
+const VERTICAL_PADDING = 16;
+const SEPARATOR_VERTICAL_PADDING = 4;
+const SEPARATOR_HEIGHT = 26;
+const TOP_LIST_PADDING = 12;
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    paddingHorizontal: utils.spacings.medium,
+}));
+
+const listStaticStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.backgroundSurfaceElevation1,
+    borderBottomLeftRadius: MANAGER_MODAL_BOTTOM_RADIUS,
+    borderBottomRightRadius: MANAGER_MODAL_BOTTOM_RADIUS,
+    borderWidth: utils.borders.widths.small,
+    borderColor: utils.colors.borderElevation0,
+    marginTop: -MANAGER_MODAL_BOTTOM_RADIUS,
+    marginBottom: -MANAGER_MODAL_BOTTOM_RADIUS,
+    paddingTop: MANAGER_MODAL_BOTTOM_RADIUS + TOP_LIST_PADDING,
+    paddingBottom: VERTICAL_PADDING,
+    zIndex: 10,
+    ...utils.boxShadows.small,
+}));
+
+export const DeviceList = ({ isVisible, onSelectDevice }: DeviceListProps) => {
+    const { applyStyle } = useNativeStyles();
+    const navigation = useNavigation<NavigationProp>();
+    const { setIsDeviceManagerVisible } = useDeviceManager();
+    const device = useSelector(selectDevice);
+    const notSelectedInstancelessDevices = useSelector(selectInstacelessUnselectedDevices);
+    const opacity = useSharedValue(0);
+    const height = useSharedValue(0);
+    const [isShown, setIsShown] = useState(false);
+
+    const handleConnectDevice = () => {
+        if (device) {
+            onSelectDevice(device);
+        }
+        setIsDeviceManagerVisible(false);
+        navigation.navigate(RootStackRoutes.ConnectDeviceStack, {
+            screen: ConnectDeviceStackRoutes.ConnectAndUnlockDevice,
+        });
+        analytics.report({
+            type: EventType.DeviceManagerClick,
+            payload: { action: 'connectDeviceButton' },
+        });
+    };
+
+    //to hide/show the list with animation
+    useEffect(() => {
+        if (isVisible) {
+            const hasNotSelectedInstancelessDevices = notSelectedInstancelessDevices.length > 0;
+
+            const paddingsHeight = VERTICAL_PADDING * 2;
+            const otherDevicesHeight =
+                notSelectedInstancelessDevices.length * (ITEM_HEIGHT + ITEMS_SPACING);
```

Comment:
nit: I don't see any benefit using these consts that are hardcoded when you can use theming sizes. 

```suggestion
                notSelectedInstancelessDevices.length * (ITEM_HEIGHT + utils.spacings.small);
```

Reply:
Most of them are only used here and it's not tied to specific paddings or other calculations, so I'd rather do just theming sizes here, WDYT?